File: backend/src/main/java/ru/fbtw/thestore/backend/datas/catalog/dto/ProductDto.java
package ru.fbtw.thestore.backend.datas.catalog.dto;

import lombok.Data;
import ru.fbtw.thestore.backend.domains.catalog.Product;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.List;
import java.util.Set;

/**
 * A DTO for the {@link Product} entity
 */
@Data
public class ProductDto implements Serializable {
	private final Long id;
	private final String productName;
	private final BigDecimal productPrice;
	private final BigDecimal productOldPrice;
	private final Integer productVolume;
	private final String productDescription;
	private final Integer productAmount;
	private final Long categoryId;
	private final List<String> pictures;
	private final Set<ProductParamDto> productParams;
}

File: mobile_app/the_store_app/lib/entity/product/product_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
part 'product_dto.freezed.dart';
part 'product_dto.g.dart';


@Freezed(
  fromJson: true,
  toJson: true,
)
class ProductDto with _$ProductDto{
  factory ProductDto({
    required int id,
    required String productName,
    required String productPrice,
    String? productOldPrice,
    required String productDescription,
    required int productAmount,
    required int categoryId,
    required List<String> pictures,
}) = _ProductDto;

  factory ProductDto.fromJson(Map<String, dynamic> json) => _$ProductDtoFromJson(json);
}